File: baekjoon/24416/24416.cpp
#include <iostream>

using namespace std;

int fibonacci_recursion(const int& n, int& callCount) {
	if (n == 1 || n == 2) {
		++callCount;
		return 1;
	}

	return fibonacci_recursion(n - 1, callCount) + fibonacci_recursion(n - 2, callCount);
}

int fibonacci_dp(const int& n, int& callCount) {
	int result[n + 1];
	result[1] = result[2] = 1;

	for (int i = 3; i <= n; ++i) {
		++callCount;
		result[i] = result[i - 1] + result[i - 2];
	}

	return result[n];
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	cout.tie(nullptr);

	int n = 0;
	cin >> n;

	int callCountRecursion = 0;
	fibonacci_recursion(n, callCountRecursion);

	int callCountDP = 0;
	fibonacci_dp(n, callCountDP);

	cout << callCountRecursion << " " << callCountDP << "\n";

	return 0;
}

File: baekjoon/24416/24416.py
import sys


def fibonacci_recursion(n, count):
    if n == 1 or n == 2:
        count[0] += 1
        return 1

    return fibonacci_recursion(n - 1, count) + fibonacci_recursion(
        n - 2, count)


def fibonacci_dp(n, count):
    result = list()
    result.append(0)
    result.append(1)
    result.append(1)

    for i in range(3, n + 1):
        count[0] += 1
        result.append(result[i - 1] + result[i - 2])

    return result[n]


if __name__ == "__main__":
    n = int(sys.stdin.readline())

    countDP = [0]
    print(fibonacci_dp(n, countDP), countDP[0])
